File: Python/problem1313.py
from typing import List

class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        ans = []
        for k in range(0, len(nums)-1, 2):
            # for _ in range(nums[k]):
            #     ans.append(nums[k+1])
            ans += nums[k] * [nums[k+1]]
        return ans

solu = Solution()
nums = [1,2,3,4]
print(solu.decompressRLElist(nums))

File: C++/problem1313.cpp
// class Solution {
// public:
//     vector<int> decompressRLElist(vector<int>& nums) {
//         map<int, int> m;
//         for (vector<int>::iterator it = nums.begin(); it < nums.end(); it += 2)
//         {
//             m.insert(pair<int, int>(*it, *(it + 1)));
//         }

//         vector<int> ans;
//         for (map<int, int>::iterator it = m.begin(); it != m.end(); ++it)
//         {
//             for (int i = 0; i < it->first; ++i)
//             {
//                 ans.push_back(it->second);
//             }
//         }

//         return ans;
//     }
// };


class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        vector<int> freqs, vals, ans;
        for (vector<int>::iterator it = nums.begin(); it < nums.end(); it += 2)
        {
            freqs.push_back(*it);
            vals.push_back(*(it + 1));
        }

        for (int i = 0; i < freqs.size(); ++i)
        {
            for (int j = 0; j < freqs[i]; ++j)
            {
                ans.push_back(vals[i]);
            }
        }

        return ans;
    }
};
